Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -439,15 +453,47 @@ bool HiresTexture::LoadTexture(Level& level, const std::vector<u8>& buffer)
   return true;
 }
 
-std::string HiresTexture::GetTextureDirectory(const std::string& game_id)
+std::set<std::string> HiresTexture::GetTextureDirectories(const std::string& game_id)
 {
+  std::set<std::string> result;
   const std::string texture_directory = File::GetUserPath(D_HIRESTEXTURES_IDX) + game_id;
 
-  // If there's no directory with the region-specific ID, look for a 3-character region-free one
-  if (!File::Exists(texture_directory))
-    return File::GetUserPath(D_HIRESTEXTURES_IDX) + game_id.substr(0, 3);
+  if (File::Exists(texture_directory))
+  {
+    result.insert(texture_directory);
+  }
+  else
+  {
+    // If there's no directory with the region-specific ID, look for a 3-character region-free one
+    const std::string region_free_directory =
+        File::GetUserPath(D_HIRESTEXTURES_IDX) + game_id.substr(0, 3);
+
+    if (File::Exists(region_free_directory))
+    {
+      result.insert(region_free_directory);
+    }
+  }
+
+  const auto match_gameid = [game_id](const std::string& filename) {
+    std::string basename;
+    SplitPath(filename, nullptr, &basename, nullptr);
+    return basename == game_id || basename == game_id.substr(0, 3);
+  };
+
+  // Look for any other directories that might be specific to the given gameid
+  const auto root_directory = File::GetUserPath(D_HIRESTEXTURES_IDX);
+  const auto files = Common::DoFileSearch({root_directory}, {".txt"}, true);
+  for (const auto& file : files)
+  {
+    if (match_gameid(file))
+    {
+      const auto directory_path = file.substr(root_directory.size());
```

Comment:
Could you please add a comment that explains what this does and why? It isn't immediately obvious what this does (get the top directory, not just the parent directory). Something similar to your explanation on IRC would be fine

Reply:
Thanks for reviewing @leoetlino .  Please take a look at the comment when you have time and see if you think it's appropriate.